src/openms_gui/source/VISUAL/SpectraIDViewTab.cpp

```diff
@@ -1001,14 +1000,7 @@ namespace OpenMS
     {
       return;
     }      
-    if (FileHandler::getTypeByFileName(filename) == FileTypes::MZIDENTML)
-    {
-      MzIdentMLFile().store(filename, prot_id, all_pep_ids);
-    }
-    else 
-    {
-      IdXMLFile().store(filename, prot_id, all_pep_ids);
-    }
+    FileHandler().storeIdentifications(filename, prot_id, all_pep_ids, {FileTypes::IDXML, FileTypes::MZIDENTML});
```

Comment:
reuse the `FileTypeList` above.
This class also has some functionality (`contains(const FileTypes::Type& type)`) which may make your code a bit more expressive.

Reply:
feel free to extend the `FileTypeList` class to abstract away more stuff from FileHandler

Reply:
Thanks! I think that's a good target for part 2 of this project.